core/domain/user_domain.py

```diff
@@ -345,3 +348,22 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected can_review_questions to be a boolean value, '
                 'found: %s' % type(self.can_review_questions))
+
+
+class ModifiableUserDataV1(python_utils.OBJECT):
+    """Domain object to represent change in allowed attributes for UserSettings
+    and UserAuthDetails domain object.
```

Comment:
Might need to update this docstring if PIN is not needed for the latter. (This would have the benefit of making this object conceptually simpler.)

Reply:
Waiting for a confusion to get cleared [here](https://github.com/oppia/oppia/pull/10241#discussion_r471706684) - I think we will mostly remove the attribute, so I will change it after getting a reply.

Reply:
Changed it now.